### PR TITLE
Missing unit test

### LINKED ISSUE #131: Missing test and docs
- [x] activity: Docs missing return value
- [x] auth: Missing Docs, Missing unit test
- [x] gist: Missing description integration test, Missing unit test
- [x] gitdata: Missing Docs, Missing integration test
- [x] issue: Doc missing return value in comments
- [x] pullrequest: Docs missing return value, Missing unit test
- [x] repo: Missing unit test
- [x] user: Missing Docs, Missing unit test

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #131 

- [x] auth: Missing unit test
- [x] gist: Missing unit test
- [x] user: Missing unit test

### CODE REVIEW

File: github4s/shared/src/test/scala/github4s/unit/AuthSpec.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        |{
+        |"client_id": "e8e39175648c9db8c280",
+        |"client_secret": "1234567890",
+        |"code": "code",
+        |"redirect_uri": "http://localhost:9000/_oauth-callback",
+        |"state": "$validAuthState"
+        |}""".stripMargin
+
+    val httpClientMock = httpClientMockPostOAuth[OAuthToken](
+      url = "http://127.0.0.1:9999/login/oauth/access_token",
```
Comment (BenFradet) [inline github4s/shared/src/test/scala/github4s/unit/AuthSpec.scala]: I'm curious as to why you had to specify the whole url?

File: github4s/shared/src/test/scala/github4s/unit/GistSpec.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        |  "public": true,
+        |  "files": {
+        |    "test.scala": {
+        |      "content": "val meaningOfLife = 42"
+        |    }
+        |  }
+        |}""".stripMargin
+
+    val httpClientMock = httpClientMockPost[Gist](
+      url = s"gists",
```
Comment (BenFradet) [inline github4s/shared/src/test/scala/github4s/unit/GistSpec.scala]: string interpolation isn't necessary

File: github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala [hunk 3]
```diff
@@ -152,38 +152,6 @@ class ReposSpec extends BaseSpec {
...
-      sampleToken,
-      headerUserAgent,
-      validRepoOwner,
-      validRepoName,
-      validTagTitle,
-      validTagTitle,
-      validNote,
-      Some("master"))
-  }
-
```
Comment (BenFradet) [inline github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala]: was this a duplicate?

Comment (AdrianRaFo) [inline github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala]: :arrow_up:

**Pull request merged**